arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -201,52 +228,15 @@ public CallSection(Hub hub, MessageFrame frame) {
     hub.defers().scheduleForPostRollback(this, currentFrame);
     hub.defers().scheduleForPostTransaction(this);
 
-    if (aborts) {
-      this.abortingCall(hub);
-      return;
-    }
-
     // The CALL is now unexceptional and un-aborted
-    hub.defers().scheduleForImmediateContextEntry(this);
-    hub.defers().scheduleForContextReEntry(this, currentFrame);
-    final WorldUpdater world = currentFrame.frame().getWorldUpdater();
-
-    if (isPrecompile(calleeAddress)) {
-      precompileAddress = Optional.of(calleeAddress);
-      scenarioFragment.setScenario(CALL_PRC_UNDEFINED);
-      // Account rows for precompile are traced at contextReEntry
-
-      precompileSubsection =
-          ADDRESS_TO_PRECOMPILE.get(preOpcodeCalleeSnapshot.address()).apply(hub, this);
-    } else {
-      Optional.ofNullable(world.get(calleeAddress))
-          .ifPresentOrElse(
-              account -> {
-                scenarioFragment.setScenario(
-                    account.hasCode() ? CALL_SMC_UNDEFINED : CALL_EOA_SUCCESS_WONT_REVERT);
-              },
-              () -> {
-                scenarioFragment.setScenario(CALL_EOA_SUCCESS_WONT_REVERT);
-              });
-
-      // TODO is world == worldUpdater & what happen if get doesn't work ?
-      //  doesn't work ?
-      // TODO: write a test where the recipient of the call does not exist in the state
-    }
-
-    if (scenarioFragment.getScenario() == CALL_SMC_UNDEFINED) {
-      this.commonValues.payGasPaidOutOfPocket(hub);
-      hub.defers().scheduleForContextReEntry(firstImcFragment, currentFrame);
-      finalContextFragment = ContextFragment.initializeNewExecutionContext(hub);
-      final boolean isSelfCall = callerAddress.equals(calleeAddress);
-      selfCallWithNonzeroValueTransfer = isSelfCall && !value.isZero();
-      hub.romLex().callRomLex(currentFrame.frame());
-      hub.defers().scheduleForContextExit(this, hub.callStack().futureId());
-    }
-
-    if (scenarioFragment.getScenario() == CALL_EOA_SUCCESS_WONT_REVERT) {
-      this.commonValues.collectChildStipend(hub);
-      finalContextFragment = ContextFragment.nonExecutionProvidesEmptyReturnData(hub);
+    refineUndefinedScenario(hub);
+    CallScenarioFragment.CallScenario scenario = scenarioFragment.getScenario();
+    switch (scenario) {
+      case CALL_ABORT_WONT_REVERT -> abortingCall(hub);
+      case CALL_EOA_UNDEFINED -> eoaProcessing(hub);
+      case CALL_PRC_UNDEFINED -> prcProcessing(hub);
+      case CALL_SMC_UNDEFINED -> smcProcessing(hub, frame);
+      default -> throw new RuntimeException("Illegal CALL scenario");
```

Comment:
Splitting up of the logic. Outside of the
```java
hub.defers().scheduleForPostRollback(this, currentFrame);
hub.defers().scheduleForPostTransaction(this);
```
all other scheduling is now done à la carte so as to simplify things.

Reply:
This is more or less the logic I had when I made the first implementation of the call, but with different class for aborting, exceptional, EOA, etc ...

Reply:
It is.